pkg/diagnose/runner.go

```diff
@@ -17,8 +18,303 @@ import (
 	"github.com/fatih/color"
 )
 
-// RunAll runs all registered connectivity checks, output it in writer
-func RunAll(w io.Writer) error {
+func init() {
+	diagnosis.Register("connectivity-datadog-autodiscovery", diagnoseMetadataAutodiscoveryConnectivity)
+}
+
+// Overall running statistics
+type diagnosisCounters struct {
+	total          int
+	success        int
+	fail           int
+	warnings       int
+	unexpected_err int
```

Comment:
Golang enforce camelCase:
```suggestion
	unexpectedErr int
```

Reply:
Done (in the next commit). Thank you.